### PR TITLE
[skyrl-train] Megatron LoRA

### PR DESCRIPTION
**Pull request by erictang000**
Enables LoRA training with the Megatron Backend. Currently waiting for https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1762 to be merged into main, so we can at least pin a commit rather than a branch for stability.

- Adds [LoRA](https://docs.nvidia.com/nemo/megatron-bridge/0.2.0/apidocs/bridge/bridge.peft.lora.html) support via Megatron-Bridge
- Adds custom checkpointing for LoRA model parameters (until LoRA checkpointing logic is upstreamed to Megatron-Bridge). 
- Weight syncing logic for Megatron + LoRA is handled by merging the LoRA parameters back into the base model before exporting to vLLM. This means that for megatron lora (for now), lora does not have to be configured for vLLM.

## Examples

GSM8K for Qwen3-30B-MoE and Qwen3-0.6B converging:
image

- Qwen3-30B-A3B previously required 2 H100 nodes for full parameter fine tuning - we can increase batch size compared to previous runs with LoRA on just 1 H100 node!

### DAPO Qwen-4B
With TIS - megatron dense backend can match/exceed FSDP backend perf. TIS is especially important for the current version of LoRA. Canonical Lora seems to be less good than "performant lora" - or maybe more sensitive to learning rate.
image

Blockers/TODOs:
- [x] ~~For Dense models, LoRA results in low grad norm/0 ppo_clip_ratio unless pp > 1. Something on megatron-core or megatron-bridge is broken for dense models.~~ Issue tracked on Megatron-Bridge (https://github.com/NVIDIA-NeMo/Megatron-Bridge/issues/1750), awaiting PR https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/1762
- [x] Test out MoE models

## Future Work
- Once Megatron-Bridge support for exporting only lora parameters is done, we should support just syncing these to vLLM for lower communication cost
- Add support for other LoRA variants from Megatron-Bridge (canonical lora, qlora, dora).

### CODE REVIEW

File: skyrl-train/tests/gpu/test_megatron_worker.py [hunk 1]
```diff
@@ -294,24 +318,127 @@ def run_hf_forward(batch, model_name):
...
+
+    #### Compare results ####
+    # compare just non-padding tokens
+    print(f"Comparing {action_log_probs_full.numel()} valid response tokens")
+    print(f"Full sample: {action_log_probs_full[:5]}")
+    print(f"Lora sample: {action_log_probs_lora[:5]}")
+
+    # max diff
+    max_diff = torch.max(torch.abs(action_log_probs_full - action_log_probs_lora))
+    print(f"Max diff: {max_diff}")
```
[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl-train/tests/gpu/test_megatron_worker.py:415]: Not sure why this comment was resolved @erictang000

[the remark this answers is not in the document]
Comment (erictang000) [reply skyrl-train/tests/gpu/test_megatron_worker.py:415]: https://github.com/NovaSky-AI/SkyRL/blob/ec738aef5e5c6bbdbd64d451ec86abe60c70bad2/skyrl-train/tests/gpu/gpu_ci/test_megatron_worker.py#L416

fixed it later
